File: app/src/main/java/com/mumanit/foursquareclient/domain/GetVenuesInteractor.java
package com.mumanit.foursquareclient.domain;

import com.mumanit.foursquareclient.domain.model.VenueData;
import java.util.List;
import rx.Observable;

/**
 * Created by pmuciek on 7/8/17.
 */

public interface GetVenuesInteractor {
    Observable<List<VenueData>> getVenues();
}

File: app/src/main/java/com/mumanit/foursquareclient/domain/GetVenuesInteractor.kt
package com.mumanit.foursquareclient.domain

import com.mumanit.foursquareclient.domain.model.VenueData
import kotlinx.coroutines.flow.Flow

interface GetVenuesInteractor {
    suspend fun getRecommendedVenues(): List<VenueData>
    fun getRecommendedVenuesWithFlow(): Flow<List<VenueData>>
}
